src/aideck/embeddings/gemini_provider.py

```
@@ -0,0 +1,665 @@
+            try:
+                if hasattr(self._credential, 'aclose'):
+                    await self._credential.aclose()
+                elif hasattr(self._credential, 'close'):
+                    if asyncio.iscoroutinefunction(self._credential.close):
+                        await self._credential.close()
+                    else:
+                        self._credential.close()
+                
+                del self._credential
+                self._credential = None
+            except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)